lib/datadog/profiling/exporter.rb

```diff
@@ -43,6 +45,9 @@ def initialize(
         @last_flush_finish_at = nil
         @created_at = time_provider.now.utc
         @internal_metadata = internal_metadata
+        # NOTE: At the time of this comment collected info does not change over time so we'll hardcode
+        #       it on startup to prevent serializing the same info on every flush.
+        @info_json = JSON.fast_generate(info_collector.info)
```

Comment:
Minor: Not that we plan to mutate it, but since it lives for a while it may be worth as a "just-in-case"

```suggestion
        @info_json = JSON.fast_generate(info_collector.info).freeze
```

(Or alternatively, maybe the `Flush` should instead do it to every one of its arguments?)